src/plugins_types.c

```diff
@@ -1073,9 +1091,38 @@ lyplg_type_resolve_leafref(const struct lysc_type_leafref *lref, const struct ly
     if (target) {
         *target = set.val.nodes[i].node;
     }
+    if (targets) {
+        for (i = 0; i < set.used; ++i) {
+            if (set.val.nodes[i].type != LYXP_NODE_ELEM) {
+                continue;
+            }
+            if (((struct lyd_node_term *)set.val.nodes[i].node)->value.realtype != value->realtype) {
+                continue;
+            }
+
+            if (!lref->plugin->compare(&((struct lyd_node_term *)set.val.nodes[i].node)->value, value)) {
+                LY_ARRAY_NEW_GOTO(LYD_CTX(node), *targets, item, rc, cleanup);
+                *item = set.val.nodes[i].node;
+            }
+        }
+    }
 
 cleanup:
     lyxp_expr_free(LYD_CTX(node), target_path);
     lyxp_set_free_content(&set);
     return rc;
 }
+
+LIBYANG_API_DEF LY_ERR
+lyplg_type_resolve_leafref(const struct lysc_type_leafref *lref, const struct lyd_node *node, struct lyd_value *value,
```

Comment:
Is this function even needed? Ignore compatibility, it is a completely new feature so it can be changed without problems.

Reply:
Well, it is called just from two places internally, which can be changed to use the second variant allowing multiple targets. Do you want me to do that?

Reply:
Yes, why not, just simplify the code and in general try to reduce the functionality instead of adding to it. Is there any point in supporting finding only a single target? I would prefer all targets are supported or only one, you have chosen the former with this PR so stick to it please.

Reply:
Done